### PR TITLE
[Feature] Manage the mapping of name to id by AuthorizationProvider instead of PrivilegeManager

### PR DESCRIPTION
**Pull request by padmejin**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Manage the mapping of name to id by AuthorizationProvider instead of PrivilegeManager, for the following reasons:
* AuthorizationProvider may add more types or actions after upgrading, but there's no way to update the new mappings in PrivilegeManager
* PrivilegeManager will persist all mappings in the image in a certain order. It's much easier if AuthorizationProvider is the one that defines the order.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 98 / 100 (98.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/privilege/DefaultAuthorizationProvider.java|25|27|92.59%|[59, 68]|
|:large_blue_circle:|com/starrocks/privilege/ActionSet.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeTypes.java|48|48|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeManager.java|23|23|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeStmtAnalyzerV2.java|1|1|100.00%|[]|

**Comment by padmejin** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/privilege/PrivilegeTypes.java [hunk 1]
```diff
@@ -2,56 +2,117 @@
 
 package com.starrocks.privilege;
 
-import java.util.List;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
+import java.util.HashMap;
+import java.util.Map;
 
 public enum PrivilegeTypes {
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/privilege/PrivilegeTypes.java:8]: why the name of enum always has an 's' in the end, it's a type not a collection i think.

**Review by gengjun-git — DISMISSED** [review]

**Review by nshangyiming — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/privilege/PrivilegeTypes.java (hunk 1 above)
Comment (padmejin) [reply fe/fe-core/src/main/java/com/starrocks/privilege/PrivilegeTypes.java]: Usted tiene razon.

**Review by nshangyiming — APPROVED** [review]

**Comment by padmejin** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 137 / 139 (98.56%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/privilege/DefaultAuthorizationProvider.java|27|29|93.10%|[59, 68]|
|:large_blue_circle:|com/starrocks/privilege/ActionSet.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/mysql/privilege/AuthUpgrader.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeType.java|56|56|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeManager.java|39|39|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeStmtAnalyzerV2.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/TablePEntryObject.java|2|2|100.00%|[]|

**Comment by padmejin** [comment]
run starrocks_admit_test

**Review by gengjun-git — APPROVED** [review]

**Comment by padmejin** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 137 / 139 (98.56%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/privilege/DefaultAuthorizationProvider.java|27|29|93.10%|[59, 68]|
|:large_blue_circle:|com/starrocks/privilege/ActionSet.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/mysql/privilege/AuthUpgrader.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeType.java|56|56|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeManager.java|39|39|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeStmtAnalyzerV2.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/TablePEntryObject.java|2|2|100.00%|[]|

**Review by meegoo — APPROVED** [review]

**Pull request merged**